homeassistant/components/sensor/history_stats.py

```diff
@@ -164,21 +168,28 @@ def update(self):
 
         # Get history between start and end
         history_list = history.state_changes_during_period(
-            self.hass, start, end, str(self._entity_id))
+            start, end, str(self._entity_id))
 
         if self._entity_id not in history_list.keys():
             return
 
         # Get the first state
-        last_state = history.get_state(self.hass, start, self._entity_id)
+        last_state = history.get_state(start, self._entity_id)
         last_state = (last_state is not None and
                       last_state == self._entity_state)
         last_time = dt_util.as_timestamp(start)
         elapsed = 0
 
         # Make calculations
         for item in history_list.get(self._entity_id):
-            current_state = item.state == self._entity_state
+            if self._attribute is not None:
+                if self._attribute not in item.attributes:
+                    _LOGGER.error("%s not in %s attributes",
+                                  self._attribute, self._entity_id)
+                    return
```

Comment:
> `if self._attribute not in item.attributes:`

You could just send a simple warning somewhere else (i.e not in the loop), and then set `current_state` to `False` if the attribute is not found, as this is not a critical error for the component.